clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/DownloadStrategyResolver.ts

```diff
@@ -50,6 +58,16 @@ export class DownloadStrategyResolver {
     copybookName: CopybookName,
     documentUri: string,
   ): Promise<boolean> {
+    const e4e = await E4ECopybookService.getE4EClient(documentUri);
+    if (this.e4eDownloader && e4e) {
```

Comment:
When do we get valid e4e and not have e4eDownloader?

Reply:
we don't. but the e4eDownloader can be undefined so this is more there to honour type check . on the other hand we can have valid downloader but not e4e client ( not a e4e file case )